Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -838,6 +1091,290 @@ void MatchTOF::doMatching(int sec)
+  auto& cacheTrk = mTracksSectIndexCache[sec];  // array of cached tracks indices for this sector; reminder: they are ordered in time!
+  int nTracks = cacheTrk.size(), nTOFCls = cacheTOF.size();
+  LOG(INFO) << "Matching sector " << sec << ": number of tracks: " << nTracks << ", number of TOF clusters: " << nTOFCls;
+  if (!nTracks || !nTOFCls) {
+    return;
+  }
+  int itof0 = 0;                          // starting index in TOF clusters for matching of the track
+  int detId[2][5];                        // at maximum one track can fall in 2 strips during the propagation; the second dimention of the array is the TOF det index
+  float deltaPos[2][3];                   // at maximum one track can fall in 2 strips during the propagation; the second dimention of the array is the residuals
+  o2::track::TrackLTIntegral trkLTInt[2]; // Here we store the integrated track length and time for the (max 2) matched strips
+  int nStepsInsideSameStrip[2] = {0, 0};  // number of propagation steps in the same strip (since we have maximum 2 strips, it has dimention = 2)
+  float deltaPosTemp[3];
+  std::array<float, 3> pos;
+  std::array<float, 3> posBeforeProp;
+  float posFloat[3];
+
+  // prematching for TPC only tracks (identify BC candidate to correct z for TPC track accordingly to v_drift)
+
+  LOG(DEBUG) << "Trying to match %d tracks" << cacheTrk.size();
+  for (int itrk = 0; itrk < cacheTrk.size(); itrk++) {
+    int nStripsCrossedInPropagation = 0; // how many strips were hit during the propagation
+    auto& trackWork = mTracksWork[cacheTrk[itrk]];
+    auto& trefTrk = trackWork.first;
+    auto& intLT = mLTinfos[cacheTrk[itrk]];
+
+    int nBCcand = 1;
+    std::vector<ulong> BCcand;
+    BCcand.reserve(nBCcand);
+
+    int side = (trefTrk.getZ() > 0) ? 1 : -1;
+
+    // look at BC candidates for the track
+    itof0 = 0;
+    double minTrkTime = (trackWork.second.getTimeStamp() - mSigmaTimeCut * trackWork.second.getTimeStampError()) * 1.E6; // minimum time in ps
+    double maxTrkTime = (trackWork.second.getTimeStamp() + mSigmaTimeCut * trackWork.second.getTimeStampError()) * 1.E6; // maximum time in ps
+    for (auto itof = itof0; itof < nTOFCls; itof++) {
+      auto& trefTOF = mTOFClusWork[cacheTOF[itof]];
+
+      if (trefTOF.getTime() < minTrkTime) { // this cluster has a time that is too small for the current track, we will get to the next one
+        itof0 = itof + 1;
+        continue;
+      }
+
+      if (trefTOF.getTime() > maxTrkTime) { // this cluster has a time that is too large for the current track, close loop
+        break;
+      }
+
+      if (trefTOF.getZ() * side < 0)
+        continue;
+
+      long bc = long(trefTOF.getTime() * Geo::BC_TIME_INPS_INV);
+
+      bool isalreadyin = false;
+
+      for (int k = 0; k < BCcand.size(); k++)
+        if (bc == BCcand[k])
+          isalreadyin = true;
+
+      if (!isalreadyin)
+        BCcand.emplace_back(bc);
+    }
+
+    for (int iBC = 0; iBC < nBCcand; iBC++) {
```

Comment:
I don't understand the reason of looping over BCs, they are actually never used.

Reply:
I started with the idea to loop on different hypoteses for BC (even if not yet implemented). I agree (as also written before) that was a very bad idea. I removed it.